protos/Makefile

```diff
@@ -1,23 +1,54 @@
 #
-#  Copyright 2018-2021 Dgraph Labs, Inc.
+# Copyright 2023 Dgraph Labs, Inc. and Contributors
 #
-#  This file is available under the Apache License, Version 2.0,
-#  with the Commons Clause restriction.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
 #
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+GOGO_PATH   := github.com/gogo/protobuf
 
-PROTO_PATH := ${GOPATH}/src:.
-PROTO_PATH := ${PROTO_PATH}:${GOPATH}/src/github.com/gogo/protobuf
+TMPDIR     := $(shell mktemp -d)
+PROTO_PATH := ${TMPDIR}/src:.
 
 .PHONY: help
 help:
 	@echo Use \"make regenerate\" to proceed.
 
+.PHONY: clean
 clean:
 	@mkdir -p api && rm -f api/api.pb.go
 
-regenerate: clean
+.PHONY: tidy-deps
+tidy-deps:
+	@go mod tidy -v
+
+.PHONY: check
+check:
+	echo "Installing proto libraries to versions in go.mod." ; \
+		go install github.com/golang/protobuf/protoc-gen-go ; \
+		go install github.com/gogo/protobuf/protoc-gen-gogofaster
+
+.PHONY: copy-protos
+copy-protos:
+	@mkdir -p ${TMPDIR}/src/${GOGO_PATH}/gogoproto
+	@cp $(shell go list -m -f "{{.Dir}}" ${GOGO_PATH})/gogoproto/gogo.proto \
+	 ${TMPDIR}/src/${GOGO_PATH}/gogoproto/gogo.proto
```

Comment:
This copy might not be necessary; in dgraph repo we need to copy three different proto files into a common directory, but here we only have one proto file.  Am I misunderstanding something?

Reply:
Done